src/main.c

```diff
@@ -103,11 +103,11 @@ static void print_usage(const char *filename)
             filename);
 }
 
-static bool parse_args(int argc, char **args)
+static bool parse_args(int *argc, char ***args)
 {
     /* parse each argument in turn */
-    for (int i = 1; i < argc; ++i) {
-        const char *arg = args[i];
+    for (int i = 1; i < *argc; ++i) {
```

Comment:
You might consider to utilize [getopt(3)](https://man7.org/linux/man-pages/man3/getopt.3.html).
Example: https://github.com/taviso/123elf/blob/main/main.c#L167

Reply:
I think `getopt ` is a cleaner way, but it does not support long option such as --quiet. If we want to parse long option, we might use `getopt_long`  but it is only compliance to GNU ( might having porting problem ), is it still recommended?

Reply:
> I think `getopt ` is a cleaner way, but it does not support long option such as --quiet. If we want to parse long option, we might use `getopt_long` but it is only compliance to GNU ( might having porting problem ), is it still recommended?

It is acceptable to handle `--quiet` with GNU extensions since this option is not part of core functionalities. You still have to denote the usage though.

Reply:
Is it also acceptable for other options? Or we could introduce a short option ( such as -q means --quiet ) to replace long option, if so we can use `getopt` seamlessly

Reply:
The latter sounds good.